### ISSUE TITLE
@primary_key and @foreign_key_type are not supported

### ISSUE THREAD

**Issue by fuelen** [bamorim/typed_ecto_schema#13]
I define my schema as follows:
```
use TypedEctoSchema
@primary_key {:id, :binary_id, read_after_writes: true}
@foreign_key_type :binary_id
```
but when I invoke `t MyApp.Post` in iex, then I see that typespecs  for primary key and foreign keys are still integer.
```

@type t() :: %MyApp.Post{
        __meta__: Ecto.Schema.Metadata.t(),
        id: integer() | nil,
        user: Ecto.Schema.belongs_to(MyApp.User.t()) | nil,
        user_id: integer() | nil,
      }
```

**Comment by bamorim** [comment]
Nice catch! I normally set my foreign key types manually so I forgot to add that one.

However, I think the `@primary_key` is working since I use it extensively.

**Comment by bamorim** [comment]
Just published `0.3.0` on hex.

Thanks!
